Data Beyond Vision essay content

thatbudakguy opened the pull request:
- Converts the DBV google doc to markdown
- Adds all required image assets for the essay
- Updates `deepzoom` and `sketchfab` shortcodes to provide images and alts for PDF rendering
- Updates `figure` shortcode to handle animated GIFs
- Updates shortcode docs
- Creates placeholder content for snippets so they can be linked from main DBV essay

thatbudakguy commented:
not sure what's going on with the build, it says: `timed out initializing value. This is most likely a circular loop in a shortcode` but I can't replicate this locally.

there's at least one lingering CSS issue; footnote 25's height is being calculated incorrectly, leading its backref to display down inside the following note. @rlskoeser if you have ideas about what's going on here that might help; I'm unfamiliar with the notes css and not sure offhand what's causing it. looks like it's related to calculating the height of the `<li>`, but it doesn't appear to be a problem in other notes with images.

thatbudakguy commented:
Ok, think the build problem is https://github.com/gohugoio/hugo/issues/4841. Posters there had similar setups to us with shortcodes that generated a lot of image derivatives. I increased the build time to 5min, and it seems to be building in a little over 30s now.

The CI workflow normally bundles the generated content in `public/` in case you want to download it, so I modified it to not bundle images, since that stage of the build was taking way too long and timed out.

rlskoeser reviewed (commented):
<s> - Still need to update concept map with the revised version in drive </s>
- The pie chart looks enormous; can we / do we want to do anything about that?
- Can we space out the modeling pictures so they aren't one after the other but have some text in between? (I see that's how they are in the doc, but it wasn't as obvious there because of the page break — and maybe also image size/layout)
- You've revised how links are displayed in the footnotes from what's in the google doc, is that something you and Grant discussed? I ask because you're changing what I thought Grant and I had agreed on.
- Can the bar chart in the weaving insights section be moved down after the first paragraph so we don't have images so close together?
- Similar question for Xinyi's charts in her insights section — can we space these out with the text? It's hard to read them as they are now
- What do you think about adding some whitespace before the acknowledgements? (I also wonder if the font size should be a little smaller)
- Do we need to preserve the `v2-01` in the filenames?
- Images in contextual notes are not displaying

rlskoeser commented:
> there's at least one lingering CSS issue; footnote 25's height is being calculated incorrectly, leading its backref to display down inside the following note. @rlskoeser if you have ideas about what's going on here that might help; I'm unfamiliar with the notes css and not sure offhand what's causing it. looks like it's related to calculating the height of the `<li>`, but it doesn't appear to be a problem in other notes with images.

I think I found and fixed this in my annotated weaving branch but I don't see a commit that I can cherry-pick. There's a max-height that's set on the contextual notes that isn't unset for the endnotes. I'll see if I can find the fix and add it to your branch.

btw are you having any trouble with the endnotes section displaying? I've been having sporadic problems with it not working on my weaving annotation branch and now I'm seeing it on this one too

thatbudakguy commented on content/issues/1/data-beyond-vision/index.md:
```diff
@@ -13,15 +13,15 @@ pdf: https://cdh.princeton.edu/media/uploads/documents/grad_fellows_call_for_spr
 
-
-How do we represent tangible objects in a visual medium? We use words, pictures, and diagrams. We describe, share, show, and fail.
+How do we represent tangible objects in a visual medium? We use words, pictures, and diagrams. We describe, share, show and fail.
```
guess we need a house ruling on oxford commas...

gwijthoff replied:
I'm usually an Oxford comma guy but think I kept it out of this one because I liked the "show and fail" echo with "show and tell." But I agree, it should be in there.

thatbudakguy commented on themes/startwords/layouts/shortcodes/deepzoom.html:
```diff
@@ -12,7 +12,12 @@
-</script>
\ No newline at end of file
+</script>
+{{/* image, alt & caption rendered in pdf only, in place of viewer */}}
+<figure class="deepzoom-preview">
+    <img src="{{ .Get `pdf-img` }}" alt="{{ .Get `pdf-alt` }}"/>
+    <figcaption><p>The online version of this essay includes an interactive deep zoom viewer displaying a high resolution capture of this object.</p></figcaption>
+</figure>
```
I didn't realize we weren't making PDFs of the snippets! Is there a reason why not? In any case, if we aren't, these can definitely be optional.

rlskoeser replied:
I don't know if we ever definitely decided on that – I think that's what @gwijthoff was leaning towards from the start, and after working more on the content I came around to that way of thinking. I'm not sure the how-tos merit the effort (reserving DOIs, generating PDFs and making sure they render properly, publishing on Zenodo). And the experimental annotated weaving doesn't work at all in a PDF.

thatbudakguy replied:
Makes sense given the process to get a DOI is kinda lengthy and manual, and we definitely don't want to discourage innovative stuff like the annotated weaving.

thatbudakguy replied:
Made it optional for both embed types. It will error if `pdf-img` is specified without `pdf-alt`.

thatbudakguy commented:
@rlskoeser I'm not having issues with the endnotes displaying except the sizing one, although the fact that we don't generate derivatives for endnotes is making lighthouse angry because we're scaling down images that are many times the size of the slot we're using. might be worth it to shrink the versions we have.

there are some a11y-related errors in CI that seem to be coming from the ability to embed links in `<figcaption>`s; the link is being created with no inner text. i'll admit I don't really understand what the figure shortcode is doing in the caption part; do we have any captions formatted like that? is it even legal to have headings inside a caption? maybe we can simplify it a little. i also don't think we should make alt optional, or have the alt use a caption if one isn't provided, since they have different use cases.

rlskoeser commented:
@thatbudakguy I thought I sized all the images for footnotes based on Gissoo's recommendation (in our image asset guideline docs). I think it's meant to be the 2x of a contextual note image — do you think they're still too big?

thatbudakguy commented:
for the image of the empty tables at the CDH, I'm seeing a ~1700px wide source image that fits into a 300px wide slot (on desktop). that seems like kind of a waste to me.

rlskoeser commented:
> for the image of the empty tables at the CDH, I'm seeing a ~1700px wide source image that fits into a 300px wide slot (on desktop). that seems like kind of a waste to me.

Ah, then I sized it wrong. Looking at the doc, Gissoo recommended a size of 896 wide, which I thought I followed. Would 600px be better? (I know the note can be wider in some cases).

Do you want me to resize, and if so which copies and what size?

thatbudakguy commented:
600px sounds perfect, since it'll cover the 300px slot width on a retina desktop. I've resized them locally and re-uploaded. Since these aren't big graphics where details matter, I think it's OK if they do end up being upscaled a little.

thatbudakguy commented on content/issues/1/data-beyond-vision/index.md:
```diff
@@ -124,104 +151,252 @@ Nick Budak, Xinyi Li
+[^17]: This is inspired in part by ethical principles from the Colored Conventions Project, which asks researchers using their data to “contextualize and narrate the conditions of the people who appear as ‘data’ and to name them when possible.” https://coloredconventions.org/about-records/ccp-corpus/
+[^18]: We wanted to include a 3D model of this physicalization in order to provide some semblance of virtually grasping and rotating the object. Because we didn’t have a way to capture the actual object in 3D, we created a model of it in Blender; this was fairly straightforward, since it consists of two simple shapes.
+[^19]: For more on the strengths and weaknesses of the pie chart, see Johnson, Steve. “The Case Against Pie Charts,” March 3, 2017. https://accelerate.uofuhealth.utah.edu/connect/steves-dojo-7-the-case-against-pie-charts
+[^20]: We do not provide a how-to for this object out of respect for the copyright of origami authors, who often spend years developing and refining single designs. The three-dimensional forms used to make this piece are a combination of two modular origami units which are the original creations of the brilliant Tomoko Fuse, and can be found in her book _Unit Origami: Multidimensional Transformat ...
```
Uploaded and linked (but the link won't work until we merge).

thatbudakguy commented:
> The pie chart looks enormous; can we / do we want to do anything about that?

It's kind of an interesting question, because when you think about it the only reason it looks enormous is because there isn't a whole lot going on in the picture...it doesn't really justify its size. I'm not sure there's actually a way to detect or prevent that, since we want the other more complex SVGs to stay big. Maybe manually specifying height?

> Can we space out the modeling pictures so they aren't one after the other but have some text in between?

Sure, I'll move the text in the "goals" section in between them.

> You've revised how links are displayed in the footnotes from what's in the google doc, is that something you and Grant discussed? I ask because you're changing what I thought Grant and I had agreed on.

Ah, my fault...I thought we needed to spell out all links so that they displayed correctly for text-only output. It looks like that might be handled by a regex already though. I can revert the ones that I changed.

> Can the bar chart in the weaving insights section be moved down after the first paragraph so we don't have images so close together?

Sure, done.

> Similar question for Xinyi's charts in her insights section — can we space these out with the text? It's hard to read them as they are now

Done.

> What do you think about adding some whitespace before the acknowledgements? (I also wonder if the font size should be a little smaller)

Sure, I added a bit more space on top. I wondered - is it still considered an `<h2>`? Should it be some other type of non-heading?

> Do we need to preserve the v2-01 in the filenames?

Probably not, but didn't seem worth the time to rename?

> Images in contextual notes are not displaying

This might be a chromium thing; they're working in FF. I will experiment more

gwijthoff commented:
The one endnote issue I've encountered comes when clicking the return arrow in the endnote to glide back to its anchored place in the text. When I then scroll to the top of the page, this is now the topmost place on the page (pictured below). I can no longer scroll any higher. Maybe not super important, but strange.

<img width="796" alt="Screen Shot 2020-10-20 at 4 17 34 PM" src="https://user-images.githubusercontent.com/5580840/96639515-cd8d3800-12ef-11eb-87de-da618b60222d.png">

thatbudakguy commented:
For the `figure` shortcode, I'd like to:
- [ ] remove the ability to make the whole thing a clickable link, since we don't currently use it
- [ ] make `alt` non-optional and don't use the caption for the alt
- [ ] remove `title` and don't render a heading in the caption
- [ ] make the caption a `<p>` which renders the caption text, attribution, and link, in that order
- [ ] document existing attributes

If we end up needing some of these features, we can always add them back.

@rlskoeser @gwijthoff thoughts?

rlskoeser commented:
@thatbudakguy proposed changes to the figure shortcode sound good to me

thatbudakguy commented:
@gwijthoff I have this exact same issue as well! super weird. only happens when you've clicked an anchor link, so the URL has a `#fragment` at the end. happens with the section navigation icons too, if you click one to go to a different section.

gwijthoff commented:
That makes sense to me on the figure shortcode changes, @thatbudakguy

gwijthoff commented:
@thatbudakguy interesting to see that you used smart quotes in your markdown. I used straight quotes in RM's essay, which Hugo "educates" when building the site, *except* in the case of a `"` before a footnote. See her [fn. 2, for example,](https://startwords.cdh.princeton.edu/issues/1/their-data-ourselves/#fnref:2) which remains a straight quote before the footnote number.

gwijthoff commented:
> The pie chart looks enormous; can we / do we want to do anything about that?
@rlskoeser agreed on the pie chart, that thing's gotta shrink, I suppose by manually specifying its size as @thatbudakguy suggests

thatbudakguy commented:
> interesting to see that you used smart quotes in your markdown
there are a couple places where they come in handy because we need to include quoted text in an html attribute, where a regular quote would terminate the attribute early unless escaped.

rlskoeser commented:
@thatbudakguy looking at the essay again, I think the bar chart in the weaving section is too big / bigger than it needs to be for the amount of information it's conveying. Maybe you can adjust it down some similar to how you're proposing to handle the pie chart?

thatbudakguy commented:
limited height for the pie and the stacked bar using new shortcode, lmk what you think

rlskoeser commented:
> limited height for the pie and the stacked bar using new shortcode, lmk what you think

they look much better!

thatbudakguy commented:
- [ ] fix links in footnotes that I changed
- [ ] figure out why lighthouse thinks our sitemap is invalid

rlskoeser commented:
@thatbudakguy thanks for restoring the links in the footnotes. Would you please also restore the dataset citations to what Grant and I came up with, as shown in the google doc? Full citation the first time (if you read in order), shortened citation after that. Telling people to "see note 16" isn't meaningful because there's no good way for them to do that (unless perhaps they're looking at the end notes), and in context it doesn't tell them anything.

thatbudakguy commented:
remaining issues have been split off, so merging this as per [slack discussion](https://cdhprinceton.slack.com/archives/CR6JCFQHE/p1603294398096200).
